Lib/test/test_shutil.py

```diff
@@ -1848,6 +1848,23 @@ def test_pathext(self):
             rv = shutil.which(program, path=self.temp_dir)
             self.assertEqual(rv, temp_filexyz.name)
 
+    @unittest.skipUnless(sys.platform == "win32", 'test specific to Windows')
+    def test_pathext_with_empty_str(self):
```

Comment:
@eryksun I copied the test from `test_pathext` (https://github.com/python/cpython/blob/master/Lib/test/test_shutil.py#L1835 ), would it make sense to just make a second `assert` there for this edge case?

Reply:
I think it's okay to duplicate the test case with a minor change, but add a comment referencing bpo-40592. 

---

For `chmod(name, S_IXUSR)`, I'd at least change it to use `S_IXUSR | S_IWUSR`. This avoids setting the readonly attribute for now, and future proofs the test just in case one day Python gets a real implementation of `chmod` and `access` for Windows. 

As implemented currently, `chmod(name, S_IXUSR)` has nothing to do with making the file executable. It just makes the file readonly (as an attribute, not a permission). Normally this would prevent the file from being deleted. That hasn't been a problem thus far because `NamedTemporaryFile` opens the file with the delete-on-close flag, which will ignore the readonly file attribute if it gets set *after the file is opened*.

Reply:
@eryksun Adjusted to your suggestion, could you please review again? Can I resolve the discussion?